### DISCUSSION TITLE
Compatibility with org-ql

### DISCUSSION THREAD

**Discussion by JulienMalka** [minad/org-modern#246] [Q&A]
Hello, 
I know that this has been discussed before, but it looks like `org-modern` is not compatible with the `org-ql` package. For some reason, `org-agenda-custom-commands` using `org-ql-block` doesn't have the right styling for TODO keywords, although the styling for the tags is correct.
I join here a screenshot of the problem and a reproducer. Not knowing who's in a better situation to understand/address this problem, I'll also open an issue in the `org-ql` repository.

![1733342582-wayshot](https://github.com/user-attachments/assets/e387cdfe-2764-4d3a-965d-9cf1aa275ce3)

```
(require 'package)
(require 'use-package)

;; Add MELPA for packages
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)

;; Make sure we install updated org from the repositories
(assq-delete-all 'org package--builtins)
(assq-delete-all 'org package--builtin-versions)


(use-package org-ql 
 :ensure t)

(use-package org
  :custom
  ;; Tell Org where to find all my stuff
  (org-directory "~/dev/org")
  (org-agenda-files '("~/dev/todos/test.org"))

  ;; My main custom agenda view
  (org-agenda-custom-commands
   '(("c" "Primary agenda view"
      (
       (org-ql-block '(and
		(todo "TODO")
		(tags "inbox")))

        (tags-todo "inbox"
        	((org-agenda-prefix-format "  %?-12t% s")
                (org-agenda-overriding-header "Inbox"))))))))


(use-package org-modern
  ;;:hook ((org-mode . org-modern-mode)
  ;;       (org-agenda-finalize . org-modern-agenda))
  :ensure t
  :config
  (global-org-modern-mode)
  :custom
  (org-modern-star 'replace)
  ;; Set manually to match leuven-theme
  (org-modern-todo-faces
   `(("TODO" :background "#DBEDFF" :foreground "#333333")
     ("NEXT" :background "#FFC8C8" :foreground "#333333")
     ("HOLD" :background "#F6FECD" :foreground "#333333")
     ("DONE" :background "#D5F1CF" :foreground "#333333"))))
     ```

**Comment by minad** [comment]
I am not an org-ql user yet, so I cannot spend much time on investigating the problem. However there has been a related problem where TODO keywords in custom agendas were not properly highlighted. See https://github.com/minad/org-modern/discussions/211. This old issue was fixed in https://github.com/minad/org-modern/commit/6550abc8c7b059cfa8b1653c280dde030787a6e7.

For reference, these are the old org-ql issues: https://github.com/minad/org-modern/issues/126, https://github.com/minad/org-modern/issues/85.

**Reply by JulienMalka** [reply]
Thank you, I've indeed read that fix and as far as I understand the fix is included in release 1.5, which is the one I am using, so I'm not sure it is relevant to my problem, or I am maybe missing something.

**Reply by minad** [reply]
It may or may not be relevant. My point here is that if the org-ql agenda is formatted/propertized in the same way as the normal custom Org agenda, then org-modern-agenda should work. But org-ql may use a different formatting.

**Comment by minad** [comment]
Discussion continues at https://github.com/alphapapa/org-ql/issues/477.
